libraries/freertos_plus/standard/https/src/iot_https_client.c

```diff
@@ -800,6 +844,53 @@ static int _httpParserOnChunkCompleteCallback(http_parser * pHttpParser)
 
 /*-----------------------------------------------------------*/
 
+static IotHttpsReturnCode_t _receiveHttpsBodyAsync(_httpsResponse_t* pHttpsResponse)
+{
+    HTTPS_FUNCTION_ENTRY(IOT_HTTPS_OK);
+    /* Get the request reference from the response for the callback function pointers. */
+    _httpsRequest_t* pHttpsRequest = pHttpsResponse->pHttpsRequest;
+
+    if(pHttpsRequest->pCallbacks->readReadyCallback)
+    {
+        /* If there is still more body that has not been passed back to the user, then this callback is invoked again. */
+        do {
+            pHttpsRequest->pCallbacks->readReadyCallback(pHttpsRequest->pUserPrivData, 
+                pHttpsResponse, 
+                pHttpsResponse->bodyRxStatus, 
+                pHttpsResponse->status);
+            if(pHttpsResponse->cancelled == true)
+            {
+                IotLogDebug("Cancelled HTTP response %d.", pHttpsResponse);
+                status = IOT_HTTPS_ASYNC_CANCELLED;
+                /* We break out of the loop and do not goto clean up because we want to print debugging logs for 
+                   the parser state and the networks status. */
+                break;
+            }
+        } while((pHttpsResponse->parserState < PARSER_STATE_BODY_COMPLETE) && (pHttpsResponse->bodyRxStatus == IOT_HTTPS_OK));
+
+        if(HTTPS_FAILED(pHttpsResponse->bodyRxStatus))
+        {
+            IotLogError("Error receiving the HTTP response body for response %d. Error code: %d",
+                pHttpsResponse,
+                pHttpsResponse->bodyRxStatus);
+        }
+
+        if(pHttpsResponse->parserState < PARSER_STATE_BODY_COMPLETE)
+        {
+            IotLogDebug("Did not receive all of the HTTP response body for response %d.", 
+                pHttpsResponse);
+        }
+    }
+
+    status = pHttpsResponse->bodyRxStatus;
+    /* This GOTO cleanup is here for compiler warnings about using HTTPS_FUNCTION_EXIT_NO_CLEANUP() without a 
+       corresponding goto. */
+    HTTPS_GOTO_CLEANUP();
```

Comment:
These are defined in iot_https_internal.h:
```
#define HTTPS_GOTO_CLEANUP()              IOT_GOTO_CLEANUP()
```

This is what the taskpool library does. Since I have some HTTP specific cleanup macros like:
```
#define HTTPS_ON_NULL_ARG_GOTO_CLEANUP( ptr )                           \
    if( (ptr == NULL) )                                                 \
    {                                                                   \
        IotLogError( #ptr " was NULL." );                               \
        IOT_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INVALID_PARAMETER );        \
    } 
```

For consistency and a cleaner look, all cleanup macros are redefined and appended by HTTPS.